ui/packages/catalog-ui-search/src/main/webapp/react-component/container/metacard-interactions/index.tsx

```diff
@@ -10,50 +10,26 @@
  *
  **/
 import * as React from 'react'
-import { Geometry } from 'wkx'
-
-import {
-  render as View,
-  Model as ViewModel,
-} from '../../presentation/metacard-interactions'
-
 import withListenTo, { WithBackboneProps } from '../backbone-container'
-
-import * as CqlUtils from '../../../js/CQLUtils'
-
-import * as CustomElements from '../../../js/CustomElements'
-
-import * as QueryConfirmationView from '../../../component/confirmation/query/confirmation.query.view'
-import * as LoadingView from '../../../component/loading/loading.view'
-import * as PopoutView from '../../../component/dropdown/popout/dropdown.popout.view'
-import * as ResultAddView from '../../../component/result-add/result-add.view'
-import * as ExportActionsView from '../../../component/export-actions/export-actions.view'
-
-import * as router from '../../../component/router/router'
-import * as sources from '../../../component/singletons/sources-instance'
-
-import * as store from '../../../js/store'
+import { hot } from 'react-hot-loader'
 import * as user from '../../../component/singletons/user-instance'
 
-import MarionetteRegionContainer from '../marionette-region-container'
+import CreateLocationSearch from './location-interaction'
+import ExpandMetacard from './expand-interaction'
+import BlacklistToggle, { isBlacklisted } from './hide-interaction'
+import DownloadProduct from './download-interaction'
+import ExportActions from './export-interaction'
+import AddToList from './add-to-list-interaction'
+import { Divider } from '../../presentation/metacard-interactions/metacard-interactions'
 
-const Query = require('../../../js/model/Query')
-const wreqr = require('wreqr')
+const plugin = require('plugins/metacard-interactions')
 
-type Props = {
+export type Props = {
   model: {} | any
-  el: {} | any
-  extensions: {} | any
-  handleShare?: () => void
-  categories: { [key: string]: Array<{}> }
+  onClose: () => void
 } & WithBackboneProps
 
-type El = {
-  find: (matcher: string) => any
-  toggleClass: (key: string, flag: boolean) => void
-}
-
-type Result = {
+export type Result = {
```

Comment:
could we make a Metacard type that has the properties we get from `get`, then map the Model's Result array into an array of these metacards? It would reduce the boilerplate code needed in the interaction files